pkg/services/ngalert/remote/remote_secondary_forked_alertmanager.go

```diff
@@ -2,35 +2,106 @@ package remote
 
 import (
 	"context"
+	"fmt"
+	"sync"
+	"time"
 
 	"github.com/grafana/grafana/pkg/infra/log"
 	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
 	"github.com/grafana/grafana/pkg/services/ngalert/models"
 	"github.com/grafana/grafana/pkg/services/ngalert/notifier"
 )
 
+//go:generate mockery --name remoteAlertmanager --structname RemoteAlertmanagerMock --with-expecter --output mock --outpkg alertmanager_mock
+type remoteAlertmanager interface {
+	notifier.Alertmanager
+	CompareAndSendConfiguration(context.Context, *models.AlertConfiguration) error
+	CompareAndSendState(context.Context) error
+}
+
 type RemoteSecondaryForkedAlertmanager struct {
 	log log.Logger
 
 	internal notifier.Alertmanager
-	remote   notifier.Alertmanager
+	remote   remoteAlertmanager
+
+	lastSync     time.Time
+	syncInterval time.Duration
 }
 
-func NewRemoteSecondaryForkedAlertmanager(l log.Logger, internal, remote notifier.Alertmanager) *RemoteSecondaryForkedAlertmanager {
-	return &RemoteSecondaryForkedAlertmanager{
-		log:      l,
-		internal: internal,
-		remote:   remote,
+type RemoteSecondaryConfig struct {
+	// SyncInterval determines how often we should attempt to synchronize
+	// state and configuration on the external Alertmanager.
+	SyncInterval time.Duration
+	Logger       log.Logger
+}
+
+func (c *RemoteSecondaryConfig) Validate() error {
```

Comment:
Shouldn't we validate the `SyncInterval`? I would have expected this to be > 0 otherwise we'll sync on every occasion and that is bad.

Reply:
I don't think we need to do that, this function gets called on every tick so there's already an interval there. Plus it's very helpful to be able to test this without waiting for the sync interval to elapse.